reactivesocket-core/src/main/java/io/reactivesocket/internal/Responder.java

```diff
@@ -348,22 +350,24 @@ public void success() {
 						@Override
 						public void error(Throwable e) {
 							RuntimeException exc = new RuntimeException(
-									"Failure outputting SetupException", e);
+								name() + ": Failure outputting SetupException", e);
 							tearDownWithError(exc);
 						}
 					});
 			}
 
 			private void tearDownWithError(Throwable se) {
 				// TODO unit test that this actually shuts things down
-				onError(new RuntimeException("Connection Setup Failure", se));
+				onError(new RuntimeException(name() + ": Connection Setup Failure", se));
 			}
 
 			@Override
 			public void onError(Throwable t) {
 				// TODO validate with unit tests
 				if (childTerminated.compareAndSet(false, true)) {
-					errorStream.accept(t);
+					if (!(t instanceof ClosedChannelException)) {
```

Comment:
@NiteshKant It's not required anymore.
@yschimke indeed, I'm not sure of any efficient way to achieve KnownErrorFilter purpose and properly dealing with subclasses without reflexion.